error: this rule takes 2 parameters but 1 parameters were supplied
 --> $DIR/rule_args_errors.rs:7:22
  |
7 |     rule too_few() = foo(1) //~ ERROR
  |                      ^^^

error: this rule takes 2 parameters but 3 parameters were supplied
 --> $DIR/rule_args_errors.rs:8:23
  |
8 |     rule too_many() = foo(1, <[_]>, 2)  //~ ERROR
  |                       ^^^

error: parameters on `pub rule` must be Rust types
  --> $DIR/rule_args_errors.rs:10:27
   |
10 |     pub rule pub_rule_arg(x: rule<()>) = "foo" //~ ERROR
   |                           ^
